docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -635,6 +635,62 @@ application behaviour at runtime.
+
+==== Filtering configuration options
+
+Configuration tracker could be instructed to exclude some of the options from the report by configuring `quarkus.config-tracking.exclude` with a comma-separated list of configuration option names that should be filtered out.
+
+==== Path values
+
+Configuration options with *absolute* path values that begin with a user home directory are, by default, transformed with Unix home directory alias '~' replacing the user home directory part and using `/` as a path element separator.
+
+This transformation could be disabled by setting `quarkus.config-tracking.use-user-home-alias-in-paths` to `false`.
+
+==== Hashing recorded configuration values
+
+Configuration values can be hashed using `SHA-512` algorithm before they are written to a file. Configuration option names whose values should be hashed can be configured in `quarkus.config-tracking.hash-options` as a comma separated list.
+
+=== Tracking build time configuration changes between builds
+
+While `quarkus.config-tracking.enabled` enables effective build time configuration report generation, there is also a way to check whether the values stored in that report have changed before the next build of the project is launched.
+
+Maven projects could add the following goal to their `quarkus-maven-plugin` configuration:
+[source,xml]
+----
+      <plugin>
+        <groupId>${quarkus.platform.group-id}</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+        <version>${quarkus.platform.version}</version>
+        <extensions>true</extensions>
+        <executions>
+          <execution>
+            <id>track-prod-config-changes</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>track-config-changes</goal>
+            </goals>
+          </execution>
+          <!-- other executions would follow below -->
+----
+
+The `track-config-canges` would be looking for `${project.basedir}/.quarkus/quarkus-prod-config-dump` by default (the file name and location could be configured) and if found, it will check whether the values stored in the config dump have now changed. It will log those options that have changed and save the current values of each of the options present in `${project.basedir}/.quarkus/quarkus-prod-config-dump` in `${project.basedir}/target/quarkus-prod-config.check` (the target file name and location can be configured). If the build time configuration has not changed since the last build both `${project.basedir}/.quarkus/quarkus-prod-config-dump` and `${project.basedir}/.quarkus/quarkus-prod-config-dump` should end up being identical.
```

Comment:
```suggestion
The `track-config-changes` goal looks for `${project.basedir}/.quarkus/quarkus-prod-config-dump` (file name and directory are configurable) and, if the file already exists, checks whether the values stored in the config dump have changed.
It will log the changed options and save the current values of each of the options present in `${project.basedir}/.quarkus/quarkus-prod-config-dump` in `${project.basedir}/target/quarkus-prod-config.check` (the target file name and location can be configured). If the build time configuration has not changed since the last build both `${project.basedir}/.quarkus/quarkus-prod-config-dump` and `${project.basedir}/.quarkus/quarkus-prod-config-dump` will be identical.
```